exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -125,79 +122,67 @@ public String toString() {
   /**
    * GT (>) predicate.
    */
-  private static <C extends Comparable<C>> LogicalExpression createGTPredicate(
-      LogicalExpression left,
-      LogicalExpression right
-  ) {
+  private static <C extends Comparable<C>> LogicalExpression createGTPredicate(LogicalExpression left, LogicalExpression right) {
     return new ParquetComparisonPredicate<C>(left, right, (leftStat, rightStat) -> {
-      // can drop when left's max <= right's min.
-      final C rightMin = rightStat.genericGetMin();
-      return leftStat.compareMaxToValue(rightMin) <= 0;
-    });
+      if (leftStat.compareMaxToValue(rightStat.genericGetMin()) <= 0) {
+        return RowsMatch.NONE;
+      }
+      return leftStat.compareMinToValue(rightStat.genericGetMax()) > 0 ? checkNull(leftStat) : RowsMatch.SOME;
```

Comment:
Why is it sufficient to check `leftStat` only? It is not obvious how `nulls` are handled, please provide comment/documentation.

Reply:
- Format only changes are reversed back. now you just can see the diffs with previous commit using git. I think this is the aim.

- abstract class removed for ParquetIsPredicate

- The hasNonNullValue simply checks if the max/min are defined in the rowgroup meta (see  Statistics.java L.240 or BooleanStatistics.java L.30)
Here we need to know if there are some nulls in the rows, but also if there are only nulls for example (like in test file blnTbl/0_0_3.parquet). This is achieved with isAllNulls.
If I replace isAllNulls with !hasNonNullValue, some tests will fail (in testBooleanPredicate for example).

- leftStat represents the predicate computed from the rowgroup meta (it is updated for each rowgroup), whereas rightStat represents the user predicate (a fix value here, like 20). Hence we need to compare the the leftStat min and max values against this unique value (20). Hence the code.

Reply:
It does not explain why is it sufficient to check `leftStats` only. There is a check for null, but in your tests, the number of nulls is zero. Even if tests pass, it is necessary to document (not only on PR, but in code comments as well) the reason.

Reply:
You are right; these tests were not proceeded.
I have created a unit test to check these specific cases : testFilterPruningWithNulls.
I had to check also the rightStat for nulls.